app/assets/javascripts/discourse/app/components/edit-topic-timer-form.js

```diff
@@ -17,73 +18,101 @@ import {
   TIME_SHORTCUT_TYPES,
   timeShortcuts,
 } from "discourse/lib/time-shortcut";
-import discourseComputed from "discourse-common/utils/decorators";
 import I18n from "discourse-i18n";
 import { FORMAT } from "select-kit/components/future-date-input-selector";
 
-export default Component.extend({
-  statusType: readOnly("topicTimer.status_type"),
-  autoOpen: equal("statusType", OPEN_STATUS_TYPE),
-  autoClose: equal("statusType", CLOSE_STATUS_TYPE),
-  autoCloseAfterLastPost: equal(
-    "statusType",
-    CLOSE_AFTER_LAST_POST_STATUS_TYPE
-  ),
-  autoDelete: equal("statusType", DELETE_STATUS_TYPE),
-  autoBump: equal("statusType", BUMP_TYPE),
-  publishToCategory: equal("statusType", PUBLISH_TO_CATEGORY_STATUS_TYPE),
-  autoDeleteReplies: equal("statusType", DELETE_REPLIES_TYPE),
-  showTimeOnly: or("autoOpen", "autoDelete", "autoBump"),
-  showFutureDateInput: or("showTimeOnly", "publishToCategory", "autoClose"),
-  useDuration: or(
-    "isBasedOnLastPost",
-    "autoDeleteReplies",
-    "autoCloseAfterLastPost"
-  ),
-  duration: null,
-  _itsatrap: null,
-
-  init() {
-    this._super(...arguments);
+export default class EditTopicTimerForm extends Component {
+  @service currentUser;
 
-    KeyboardShortcuts.pause();
-    this.set("_itsatrap", new ItsATrap());
-
-    this.set("duration", this.initialDuration);
-  },
+  @tracked timerType;
 
-  get initialDuration() {
-    if (!this.useDuration || !this.topicTimer.duration_minutes) {
+  duration = (() => {
```

Comment:
dunno if this pattern is good 😉 

I think I like it better than the previous

```js
duration = this.initialDuration;

get initialDuration() {
  // a getter that's used only above
}
```

but maybe it'd be cleaner to just do this in the constructor 🤷

Reply:
Yeah agreed - the current pattern is cool, but also very surprising!

But also... having a getter feels weird. So maybe a function instead? `calculateInitialDuration()`?

```js
duration = this.#calculateInitialDuration();

#calculateInitialDuration(){
  //...
}
```

?

Reply:
lol, the prop was unused 😄 the best code is deleted code 😛